adaptors/library/jobs/satusehat-create-medication.js

```diff
@@ -0,0 +1,115 @@
+// Create a medication from visit data and lookup table
+
+fn(state => {
+  // state.visit is a Commcare Visit record
+  const properties = state.visit.properties;
+  const medicineKeys = extractMedicineKeys(properties);
+
+  const obatList = state.lookup_table.obat;
+
+  state.medicationSatusehatId = [];
+
+  // Map medication data for each relevant key
+  state.medicationMappedData = mapMedications(
+    medicineKeys,
+    properties,
+    obatList, // obatList is the lookup table from Commcare
+    state,
+    util
+  );
+
+  console.log(`${state.medicationMappedData.length} Medications created`);
+  return state;
+});
+
+// Function to extract the relevant medicine keys
+function extractMedicineKeys(properties) {
+  return Object.keys(properties).filter(
+    key =>
+      key.startsWith('prescription_') &&
+      !key.includes('_amount') &&
+      !key.includes('_instruction') &&
+      !key.includes('_dose')
+  );
+}
+
+// Function to map the medications to FHIR Medication resource format
+function mapMedications(medicineKeys, properties, obatList, state, util) {
+  return medicineKeys
+    .flatMap(medicineKey => {
+      const medicineName = properties[medicineKey];
+
+      // Filter obat list to find matching medicine by name
+      return obatList.filter(item => item.fields['Nama'] === medicineName);
+    })
+    .map(item => {
+      if (item.fields.satusehat_id) {
+        // If satusehat_id is present, add to medicationSatusehatId array
+        state.medicationSatusehatId.push({
+          [`prescription_${state.medicationSatusehatId.length + 1}`]:
+            item.fields.satusehat_id,
+        });
+        return null;
+      } else {
+        // Create a new Medication resource if no satusehat_id exists
+        state.medicationId = util.uuid();
+        state.medicationSatusehatId.push({
+          [`prescription_${state.medicationSatusehatId.length + 1}`]:
+            state.medicationId,
+        });
+        return createMedicationResource(item, state.medicationId);
+      }
+    })
+    .filter(data => data !== null);
+}
+
+function createMedicationResource(item, medicationId) {
+  return {
+    fullUrl: `urn:uuid:${medicationId}`,
+    request: {
+      method: 'POST',
+      url: 'Medication',
+    },
+    resource: {
+      resourceType: 'Medication',
+      id: state => state.medicationId,
+      meta: {
+        profile: [
+          'https://fhir.kemkes.go.id/r4/StructureDefinition/Medication',
+        ],
+      },
+      identifier: [
+        {
+          system: `http://sys-ids.kemkes.go.id/medication/${state.asri_satusehat_id}`,
```

Comment:
Weird that this is using state. Looks like a bug? Maybe we need to pass this value into the function?